src/core/tsi/s2a/handshaker/s2a_handshaker_client.cc

```diff
@@ -16,41 +16,257 @@
  *
  */
 
+#include <grpc/support/port_platform.h>
+
 #include "src/core/tsi/s2a/handshaker/s2a_handshaker_client.h"
 
+#include <grpc/support/alloc.h>
+#include <grpc/support/log.h>
+
+#include "src/core/lib/slice/slice_internal.h"
+#include "src/core/lib/surface/call.h"
+#include "src/core/lib/surface/channel.h"
+#include "src/core/tsi/s2a/handshaker/s2a_handshaker_util.h"
+#include "src/core/tsi/s2a/s2a_constants.h"
+#include "src/proto/grpc/gcp/s2a.upb.h"
+
 namespace grpc_core {
 namespace experimental {
 
-tsi_result s2a_handshaker_client_start_client(
-    const s2a_handshaker_client* client) {
-  return TSI_UNIMPLEMENTED;
+/** ------------ Preparation of client start messages. --------------- **/
+
+grpc_byte_buffer* S2AHandshakerClient::SerializedStartClient() {
```

Comment:
This function seems too long and could you please add some comments between lines (based on your judgement) to make it more readable?

Reply:
Done.